test/metabase/driver/common/parameters/dates_test.clj

```diff
@@ -130,98 +129,281 @@
-                                           "today"     {:start "2016-06-07", :end "2016-06-07"}}
-             "relative (past) with starting from" {"past1days-from-0days"      {:start "2016-06-06", :end "2016-06-06"}
-                                                   "past1months-from-0months"  {:start "2016-05-01", :end "2016-05-31"}
-                                                   "past1months-from-36months" {:start "2013-05-01", :end "2013-05-31"}
-                                                   "past1years-from-36months"  {:start "2012-01-01", :end "2012-12-31"}
-                                                   "past3days-from-3years"     {:start "2013-06-04", :end "2013-06-06"}}
-             "relative (next) with starting from" {"next2days-from-1months"    {:start "2016-07-08", :end "2016-07-09"}
-                                                   "next1months-from-0months"  {:start "2016-07-01", :end "2016-07-31"}
-                                                   "next1months-from-36months" {:start "2019-07-01", :end "2019-07-31"}
-                                                   "next1years-from-36months"  {:start "2020-01-01", :end "2020-12-31"}
-                                                   "next3days-from-3years"     {:start "2019-06-08", :end "2019-06-10"}
-                                                   "next7hours-from-13months"  {:start "2017-07-07T13:00:00", :end "2017-07-07T19:00:00"}}}]
-      (testing group
-        (doseq [[s inclusive-range]   s->expected
-                [options range-xform] (letfn [(adjust [m k amount]
-                                                (if-not (get m k)
-                                                  m
-                                                  (update m k #(u.date/format (u.date/add (u.date/parse %) :day amount)))))
-                                              (adjust-start [m]
-                                                (adjust m :start -1))
-                                              (adjust-end [m]
-                                                (adjust m :end 1))]
-                                        {nil                                              identity
-                                         {:inclusive-start? false}                        adjust-start
-                                         {:inclusive-end? false}                          adjust-end
-                                         {:inclusive-start? false, :inclusive-end? false} (comp adjust-start adjust-end)})
-                :let                  [expected (range-xform inclusive-range)]]
-          (is (= expected
-                 (params.dates/date-string->range s options))
-              (format "%s with options %s should parse to %s" (pr-str s) (pr-str options) (pr-str expected))))))))
+            {"absolute datetimes"         {"Q1-2016"               [{:start "2016-01-01" :end "2016-03-31" :unit :quarter}  ;; inclusive start + end = true (default)
```

Comment:
changing the format of this to observer the results when changing the options clearer.